### PR TITLE
Validate norm_type in lp_pool1d/2d/3d

### LINKED ISSUE #134841: LPPool2d lacks a check for the validity of norm_type
### 🐛 Describe the bug

```python
import torch

x = torch.randn(1, 3, 224, 224)
p = torch.nn.LPPool2d(norm_type=0.0, kernel_size=2)
print(p(x))
```
When the norm_type of torch.nn.LPPool2d is set to 0, a division by zero error occurs during the computation. Because the norm_type parameter is used to specify the value of p in the Lp norm, when p=0, the calculation formula will include a division by zero, causing the program to crash.
In my opinion, the following improvements can be made:

1. **Parameter Check**: Add a check for `norm_type` before creating the `LPPool2d` layer to ensure it is not set to 0.

2. **Improve Documentation and Error Messages**: Update the `LPPool2d` documentation to clearly state that `norm_type` cannot be 0, or add a check in the code implementation to throw a more meaningful error message earlier.

### Versions

torch 2.0.0

### PR DESCRIPTION
**Pull request by qflen**
Fixes #134841.

## Summary

`F.lp_pool1d/2d/3d` compute `(avg_pool(x**p))**(1/p)`; with `norm_type=0` the `1/p` divides by zero and the result is NaN/Inf instead of a useful error. The `nn.LPPool*` modules forward to these and inherit the bug.

Raises `ValueError` at the top of each functional when `norm_type == 0`. Check is guarded by `isinstance(norm_type, (int, float))` so FX `symbolic_trace` (which passes `Proxy` for every arg) skips it and still hits the existing `Proxy object cannot be iterated` trace error that `test_nn_functional_lp_pool2d`/`3d` expect.

Picks up #143310 (stale) and works around the FX-tracing regression that blocked it.

## Test plan

`TestPoolingNN.test_lp_pool_rejects_zero_norm_type` parametrized over `dim ∈ {1, 2, 3}`.

### CODE REVIEW

File: torch/nn/functional.py [hunk 1]
```diff
@@ -1095,6 +1095,8 @@ def lp_pool3d(
 
     See :class:`~torch.nn.LPPool3d` for details.
     """
+    if isinstance(norm_type, (int, float)) and norm_type == 0:
```
Comment (albanD) [inline torch/nn/functional.py:1099]: norm_type is only ever supposed to be these types. Did you meant to do not + or here?

Comment (qflen) [reply torch/nn/functional.py:1099]: The `isinstance` check does look redundant from the annotation alone, but it is needed for `fx.symbolic_trace`. During tracing, `fx` passes a `Proxy` for every argument, including `norm_type`. Without this guard, the new `ValueError` gets raised on the proxy and ends up masking the existing `TraceError` (`Proxy object cannot be iterated`) that `test_nn_functional_lp_pool2d/3d` rely on in `test/test_fx.py` under `UNTRACEABLE_FUNCTIONALS -> PROXY_ITERATED`.

Rewriting it as `not isinstance(...) or norm_type == 0` runs into the same issue, since `Proxy` is neither `int` nor `float`.

I'll add a one-line comment in the code so the intent is clear to the next reader.

Comment (albanD) [reply torch/nn/functional.py:1099]: That sounds surprising. That would mean that all other type checks would fail?

Comment (qflen) [reply torch/nn/functional.py:1099]: It doesn't generalize because most other functionals already break fx tracing for structural reasons (`_pair`/`_triple` iteration, `len()`, etc.) before any scalar check would matter. `lp_pool1d` is one of the few that doesn't.

What shifts if I drop the guard and just write `if norm_type == 0:`:

| function | main today | plain `if norm_type == 0:` |
|---|---|---|
| `lp_pool2d` | untraceable, `PROXY_ITERATED` (pinned in `test_fx.py`) | untraceable, `CONTROL_FLOW`, pin needs updating |
| `lp_pool3d` | untraceable, `PROXY_ITERATED` (pinned) | untraceable, `CONTROL_FLOW`, pin needs updating |
| `lp_pool1d` | traces fine (not in `UNTRACEABLE_FUNCTIONALS`) | becomes untraceable (`CONTROL_FLOW`), new pin needed |

The guard keeps `lp_pool1d` traceable and avoids two `test_fx.py` churns.

**Comment by qflen** [comment]
Hi @albanD, does my Apr 22 reply address the concern? 

TL;DR the isinstance guard exists specifically because lp_pool1d is the one LP-pool functional that's still FX-traceable today.

Dropping the guard would force two test_fx.py pins (lp_pool2d, lp_pool3d) to flip from PROXY_ITERATED to CONTROL_FLOW and add a new pin for lp_pool1d.

**Comment by qflen** [comment]
Closing, superseded by 41e4dc9fd51.
